std/concurrency.d

```diff
@@ -1986,8 +1988,15 @@ private
                             return true;
                         }
                     }
+                    else
+                    {
+                        if (i + 1 == Ops.length)
+                        {
+                            return false;
+                        }
```

Comment:
@Sandman83 why did you add this check?
`i` can never be bigger than `Ops.length - 1`...

Reply:
If I remember correctly, compiler complained about, that the function doesn't return a value otherwise

Reply:
That's weird and shouldn't be the case. Do you have a test case?

Reply:
This was the case while compiling with -profile, if I remember correctly.
Can't reproduce it right now, however...